### ISSUE TITLE
No API token provided. You need to set the REPLICATE_API_TOKEN environment variable or create a client with `replicate.Client(api_token=...)`.

### ISSUE THREAD

**Issue by purerabbit** [replicate/replicate-python#58]

**Comment by Dewey7171** [comment]
It just seems to depend on where you use the parameters.

import webbrowser

import replicate

api = replicate.Client(api_token='TOKEN')
model = api.models.get("stability-ai/stable-diffusion")
output_url = model.predict(prompt="electric sheep, neon, synthwave")[0]
print(output_url)
webbrowser.open(output_url)

**Comment by hbarnard** [comment]
Thanks, this is a completely different approach to my current one, so I'll give it a spin.

---------
https://www.hughbarnard.org

***@***.***
Mastodon: @***@***.***
Book:http://tinyurl.com/2p9v7ba7

------- Original Message -------

**Comment by hbarnard** [comment]
> It just seems to depend on where you use the parameters.
> 
> import webbrowser
> 
> import replicate
> 
> api = replicate.Client(api_token='TOKEN') model = api.models.get("stability-ai/stable-diffusion") output_url = model.predict(prompt="electric sheep, neon, synthwave")[0] print(output_url) webbrowser.open(output_url)

Thanks I've combined your method with my approach and this works, for at least one of my problems:
       
```
        load_dotenv()
        ...
        api = replicate.Client(api_token=os.environ.get('REPLICATE_API_TOKEN'))
        model = api.models.get("j-min/clip-caption-reward")
        image_file = Path(file_path)
        result = model.predict(image=image_file)

```

**Comment by Robouste** [comment]
```python
import replicate
import os;

os.environ["REPLICATE_API_TOKEN"] = "<token>"

model = replicate.models.get("cjwbw/anything-v3.0")
version = model.versions.get("f410ed4c6a0c3bf8b76747860b3a3c9e4c8b5a827a16eac9dd5ad9642edce9a2")
```

That's what worked for me using Windows. Setting the token in the windows environment variables didn't work.

**Comment by hbarnard** [comment]
> ```python
> import replicate
> import os;
> 
> os.environ["REPLICATE_API_TOKEN"] = "<token>"
> 
> model = replicate.models.get("cjwbw/anything-v3.0")
> version = model.versions.get("f410ed4c6a0c3bf8b76747860b3a3c9e4c8b5a827a16eac9dd5ad9642edce9a2")
> ```
> 
> That's what worked for me using Windows. Setting the token in the windows environment variables didn't work.

I'm using Linux Mint. For the same user and script usage directly, I can pick up from the environment, no problem. If I use Python `subprocess` (which I want) there's a problem. I've ended up with a little hidden file with my token so it stays out of github.

**Comment by a-toms** [comment]
The problem for me was a misleading error message of: `Replicate.exceptions.ReplicateError: You didn't pass an authentication token. Learn how to authenticate and get your API token here`

In reality,  I hadn't set a version name, and was calling `version = model.versions.get()` with a argument of an empty string.

**Comment by mattt** [comment]
Hi everyone. Sorry for any inconvenience caused by this behavior. This should be fixed in versions >= 0.15.0. If you're still having problems with Replicate's default client (used in top-level `replicate.run`, et al.), you can create your own `Client` and pass an API token explicitly:

```python
from replicate.client import Client

replicate = Client(api_token="r8_.......")
```
